benchmark/schedulers/schedulers.jl

```diff
@@ -0,0 +1,58 @@
+using BenchmarkTools, Agents
+
+const SCHED_SUITE = BenchmarkGroup(["Schedulers"])
+
+include("setup.jl")
+
+for (scheduler, name) in [
+        (Schedulers.fastest, "fastest"),
+        (Schedulers.by_id, "by_id"),
+        (Schedulers.ByID(), "ByID"),
+        (Schedulers.randomly, "randomly"),
+        (Schedulers.Randomly(), "Randomly"),
+        (Schedulers.by_property(:group), "by_property"),
+        (Schedulers.ByProperty(:group), "ByProperty"),
+    ]
+    model, agent_step!, model_step! = schelling_with_scheduler(; scheduler)
+    SCHED_SUITE[name] = @benchmarkable step!($model, $agent_step!, $model_step!)
```

Comment:
Sorry for taking so long, I did an eye surgery. I can see barely now, I have 200% zoom on firefox. Alrighty, so, here is the thing. These benchmarks are inappropriate. They do not time the amount of time necessary to do scheduling, which is what their name suggests. They time the amount of time necessary to do schedulling and stepping. Now, if stepping takes 99.9% of time, then even if you've made schedulling 100x faster, it will barely show in the benchmarks. Furthermore, it is just in general good practice to benchmark _only_ the thing you care about. Functions such as nearby agents (which are in `step!` here) are benchmarked elsewhere.

So, can you please do the appropriate changes to the benchmarks? Given that in the end of the day all benchmark operations are `scheduler(model)`, you can reduce the source code of this file a lot as well via loop :)

Reply:
I really hope you're okay, and that you get better soon.

I haven't had the time to look at this either, since I have exams coming up shortly. I'll try to get to it in a week or so, when they're mostly over. Thanks for pointing out the issues here, I guess I have stuff to learn when it comes to writing benchmarks! :P I'll fix these as soon as I can.

Reply:
Things are okay, I had some small complications during the surgery and have an inflammation, but the doctor says nothing to worry. But then again I don't trust doctors :D but all is well :)